### ISSUE TITLE
`opkg` does not use it's config file.

### ISSUE THREAD

**Issue by drawkula** [minoca/os#41]
```
(root@minoca:pty8)~# opkg list-installed
```
...no output.
```
(root@minoca:pty8)~# opkg -f /etc/opkg/opkg.conf list-installed | wc -l
     77
```
Without telling `opkg` which config file to use, it seems to fail.
(The `wc` part just keeps the output of the example short.)
```
(root@minoca:pty8)~# opkg install /src/x86rel/bin/packages/vim_7.4_minoca-i686.ipk 
Not selecting vim 7.4 due to incompatible architecture.
Unknown package 'vim'.
Collected errors:
 * opkg_install_cmd: Cannot install package vim.
(root@minoca:pty8)~# opkg -f /etc/opkg/opkg.conf install /src/x86rel/bin/packages/vim_7.4_minoca-i686.ipk
Installing vim (7.4) on root.
Configuring vim.
```
...so without config file it does not even know the architecture and won't accept packages for `minoca-i686`. `opkg` defaults to only acceppt architecture `all` by default. Again, it works as expected when adding `-f /etc/opkg/opkg.conf`.

Can someone confirm this behaviour or do I have a local problem here?

**Comment by melezhik** [comment]
+1 for this, on freshly build minoca image I need to say explicitly:

`opkg -f /etc/opkg/opkg.conf`

or I can't have my opkg works properly

**Comment by drawkula** [comment]
I currently need to use `-f /etc/opkg/opkg.conf` with each call or even longer stuff to define the arch `opkg` shall accept: `opkg --add-arch minoca-i686:100 [other args and opts]`.

Using `opkg -f /etc/opkg/opkg.conf` seems to be the shortest way for now.

If all else fails, making this an `alias` in the shell's startup file will help.

**Comment by evangreen** [comment]
Are you using an opkg that you built from source or an extracted ipk of opkg that you pulled off our website? If you built it from source, what OS are you building on?

It sounds like opkg isn't looking in the right place for its config file. If you've got a kernel debugger hooked up, you could try setting IoDebugPrintOpens to TRUE (break in to the debugger with Control+B and run `eb IoDebugPrintOpens 1`, then run `g` to continue). That might give you a hint as to what file opkg is trying to open to get its config.

I can also try to look into this given the answers to the above questions.

**Comment by drawkula** [comment]
I hoped for feedback of other users whether they see the same behaviour on self built `x86rel-0.3.0-2fa9de8` images.

Here a cross built and natively built `opkg` behaved in the same way.

Having killed my image somehow is a possibility.

**Comment by ccstevens** [comment]
Fixed with [771d8d60](https://github.com/minoca/os/commit/771d8d60349960f6f69c1041ec90bd298c40761d).

**Comment by drawkula** [comment]
@ccstevens: See below...

![20161109-041827-gmt--minoca-opkg-now-worky](https://cloud.githubusercontent.com/assets/9768622/20127137/3097da74-a63c-11e6-9530-35cca45fa5e5.png)
 
PARTY!!!
